### ISSUE TITLE
subrepo branches can conflict with main repo hooks

### ISSUE THREAD

**Issue by ingydotnet** [ingydotnet/git-subrepo#135]
if they don't expect subrepo content.

See #133

Exporting an environment variable for git-subrepo indicating subrepo
interaction can give hooks something to change their behavior.

@blucz can you comment on the hooks that subrepo can interfere with?

Relevant doc changes will be necessary.

If we wanted to go the extra mile, we git-subrepo could detect user hooks that
didn't contain the env var. I'm sure that gets messy with edge cases and such
though.

**Comment by blucz** [comment]
Our hooks part of a mechanism that we built for managing binaries outside of the repository. The hooks invoke the tool, which  checks for a special file in the root of the repository related to the files it's tracking and "exit 1"s if it's not there.

FWIW we decided to fix this by having the tool (not the hooks) behave gracefully during subrepo operations, since the hooks are just one-liners that invoke the tool, so checking for the env var in the hooks might not be as general a solution as it seems at first.

I'm sure there are many hooks that are totally benign to git subrepo, too. Ours just happened to have a sanity check that caused it to experience a blanket failure.
